src/query/storages/parquet/src/pruning.rs

```diff
@@ -277,6 +276,45 @@ impl PartitionPruner {
             stats.read_rows += sub_stats.read_rows;
         }
 
+        if max_compression_ratio <= 0.0 || max_compression_ratio >= 1.0 {
+            max_compression_ratio = 1.0;
+        }
+        if max_compressed_size == 0 {
+            let max = 128usize >> 20;
+            max_compressed_size = (max as f64 / max_compression_ratio) as u64;
+        }
+
+        stats.read_rows += small_files.len();
+        let mut num_small_files = small_files.len();
+        let mut small_part = vec![];
+        let mut part_size = 0;
+        let mut make_small_files_part = |files, part_size| {
+            let estimated_uncompressed_size = (part_size as f64 / max_compression_ratio) as u64;
+            partitions.push(ParquetPart::SmallFiles(ParquetSmallFilesPart {
+                files,
+                estimated_uncompressed_size,
+            }));
+            stats.partitions_scanned += 1;
+            stats.partitions_total += 1;
+            num_small_files -= 1;
+        };
+        let max_files = self.columns_to_read.len() * 2;
+        for (path, size) in small_files.into_iter() {
+            stats.read_bytes += size as usize;
+            if !small_part.is_empty()
+                && (part_size + size > max_compressed_size || small_part.len() + 1 >= max_files)
+            {
+                make_small_files_part(mem::take(&mut small_part), part_size);
+                part_size = 0;
+            }
+            small_part.push((path, size));
+            part_size += size;
+        }
+        if !small_part.is_empty() {
+            make_small_files_part(mem::take(&mut small_part), part_size);
+        }
+        assert_eq!(num_small_files, 0);
```

Comment:
Better to add some comments to describe these codes.

Reply:
added a commit, plz review again, thank you.